src/notebook.c

```diff
@@ -430,6 +430,29 @@ static void on_open_in_new_window_activate(GtkMenuItem *menuitem, gpointer user_
 }
 
 
+static gboolean has_tabs_on_right(GeanyDocument *doc)
+{
+	GtkNotebook *nb = GTK_NOTEBOOK(main_widgets.notebook);
+	gint total_pages = gtk_notebook_get_n_pages(nb);
+	gint doc_page = document_get_notebook_page(doc);
+	return total_pages > (doc_page + 1);
+}
+
+
+void on_close_documents_right_activate(GtkMenuItem *menuitem, GeanyDocument *doc)
+{
+	g_return_if_fail(has_tabs_on_right(doc));
+	GtkNotebook *nb = GTK_NOTEBOOK(main_widgets.notebook);
+	gint doc_page = document_get_notebook_page(doc);
+	for (gint i = doc_page + 1; i < gtk_notebook_get_n_pages(nb); )
+	{
+		if (! document_close(document_get_from_page(i)))
+			i++; // only increment if tab wasn't closed
+	}
+	gtk_notebook_set_current_page(nb, doc_page);
```

Comment:
This should probably not go to the document's page but to the originally current one if it hasn't been closed; because like that using the feature on a non-current tab makes that tab current, even if the previously active tab was on the left of the one on which you activated the feature.